arangod/Transaction/Methods.cpp

```diff
@@ -1173,6 +1173,9 @@ void transaction::Methods::invokeOnAllElements(std::string const& collectionName
 
   TRI_voc_cid_t cid = addCollectionAtRuntime(collectionName);
```

Comment:
`addCollectionAtRuntime(..)` throws a not found exception

Reply:
Not in its overloaded variant.

Reply:
https://github.com/arangodb/arangodb/blob/devel/arangod/Utils/SingleCollectionTransaction.h#L66

Reply:
perhaps it would be more efficient to throw in the overloaded variant then ?

Reply:
Not sure how to detect an error here given the current implementation of the overloaded variant:
```
 64   /// @brief add a collection to the transaction for read, at runtime
 65   /// note that this can only be ourselves
 66   TRI_voc_cid_t addCollectionAtRuntime(std::string const&) override final {
 67     return _cid;
 68   }
```
If we added full collection name resolution here, there will not be any need for the overloaded variant, and additionally, it would potentially get a lot slower. So I would first like to try where this adjusted error handling takes us and if there are any errors caught by it.
I am currently trying to debug some test failures in devel fail when using a collection/shard not known to a transaction. This seems to be a very rare case and not easy to reproduce, so any debugging here helps.

Reply:
just add a maintenance mode sanity check against `_trxCollection->collectionName()` ?

Reply:
ok, trying it...